torch/utils/hipify/cuda_to_hip_mappings.py

```diff
@@ -8568,6 +8568,7 @@
 
 CAFFE2_SPECIFIC_MAPPINGS = collections.OrderedDict(
     [
+        ("PYTORCH_NO_CUDA_MEMORY_CACHING", ("PYTORCH_NO_CUDA_MEMORY_CACHING", API_CAFFE2)),
```

Comment:
If no change is needed, why do we still need this item? Do we mean PYTORCH_NO_HIP_MEMORY_CACHING as the value instead?

Reply:
The `CAFFE2_SPECIFIC_MAPPINGS` is ordered.  We want to not change the env var name here.  I've added both env var variants to this PR.  It is confusing to our users when to use "CUDA" or "HIP".  We try our best to accommodate them coming from a script that worked for cuda and it should work similarly for amd gpus on hip.  If the above mapping didn't exist, the env var gets renamed to the "HIP" one but we want both env vars to be acceptable.